### PR TITLE
[fix] Add forgotten _check_array to IncrementalBasicStatistics.partial_fit

### PR DESCRIPTION
**Pull request by olegkkruglov**
### Description

* `_check_array` is used now in `IncrementalBasicStatistics.partial_fit`. It prevents failure in case if non-contiguous numpy array has been provided as argument. 

---

Checklist to comply with before moving PR from draft:

**PR completeness and readability**

- [x] I have reviewed my changes thoroughly before submitting this pull request.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have updated the documentation to reflect the changes or created a separate PR with update and provided its number in the description, if necessary.
- [x] Git commit message contains an appropriate signed-off-by string _(see [CONTRIBUTING.md](https://github.com/intel/scikit-learn-intelex/blob/main/CONTRIBUTING.md#pull-requests) for details)_.
- [x] I have added a respective label(s) to PR if I have a permission for that.  
- [x] I have resolved any merge conflicts that might occur with the base branch.

**Testing**

- [x] The unit tests pass successfully.
- [x] I have run it locally and tested the changes extensively.

**Performance**

- [x] I have measured performance for affected algorithms using [scikit-learn_bench](https://github.com/IntelPython/scikit-learn_bench) and provided at least summary table with measured data, if performance change is expected.
- [x] I have provided justification why performance has changed or why changes are not expected.

### CODE REVIEW

**Review by samir-nasibli — APPROVED** [review]

**Comment by samir-nasibli** [comment]
Assuming green CI. I am good with the proposed changes

**Review by Alexandr-Solovev — APPROVED** [review]

File: onedal/basic_statistics/incremental_basic_statistics.py [hunk 1]
```diff
@@ -96,6 +97,12 @@ def partial_fit(self, X, weights=None, queue=None):
         policy = self._get_policy(queue, X)
         X, weights = _convert_to_supported(policy, X, weights)
 
+        X = _check_array(X, dtype=[np.float64, np.float32], ensure_2d=False)
```
Comment (icfaust) [inline onedal/basic_statistics/incremental_basic_statistics.py:100]: https://github.com/intel/scikit-learn-intelex/blob/main/sklearnex/basic_statistics/incremental_basic_statistics.py#L143 https://github.com/scikit-learn/scikit-learn/blob/main/sklearn/utils/validation.py#L2109  _check_sample_weight and the checks on X both do finite value checking.  This addition duplicates the finite value checking, ideally finite value checking should be disabled here.

**Review by icfaust — COMMENTED** [review]
Small change necessary before merge.

File: onedal/basic_statistics/incremental_basic_statistics.py (hunk 1 above)
Comment (icfaust) [reply onedal/basic_statistics/incremental_basic_statistics.py:100]: I really need to think of a way to verify finite value checking to only occur once in development.

**Comment by olegkkruglov** [comment]
https://intel-ci.intel.com/ef69d675-b999-f16b-9b9b-a4bf010d0e2e

**Review by icfaust — APPROVED** [review]
Ready to go, thanks @olegkkruglov
